dataclients/kubernetes/ingress.go

```diff
@@ -351,11 +360,47 @@ func hasCatchAllRoutes(routes []*eskip.Route) bool {
 	return false
 }
 
+// addHostTLSCert adds a TLS certificate to the certificate registry when the referenced secret is found
+// and is a valid TLS secret.
+func addHostTLSCert(ic ingressContext, hosts []string, secretName string, ns string) error {
+	for _, secret := range ic.state.secrets {
```

Comment:
`secrets` is a map, it should lookup namespace+name instead of iterating

Reply:
Using a ResourceID in map to find secrets now